relayer/relays/parachain/main.go

```diff
@@ -2,139 +2,93 @@ package parachain
 
 import (
 	"context"
-	"fmt"
 
 	"golang.org/x/sync/errgroup"
 
-	"github.com/sirupsen/logrus"
 	"github.com/snowfork/snowbridge/relayer/chain/ethereum"
 	"github.com/snowfork/snowbridge/relayer/chain/parachain"
 	"github.com/snowfork/snowbridge/relayer/chain/relaychain"
 	"github.com/snowfork/snowbridge/relayer/crypto/secp256k1"
+
+	log "github.com/sirupsen/logrus"
 )
 
-type Worker struct {
-	parachainConfig       *parachain.Config
-	relaychainConfig      *relaychain.Config
-	ethereumConfig        *ethereum.Config
+type Relay struct {
+	config                *Config
 	parachainConn         *parachain.Connection
 	relaychainConn        *relaychain.Connection
 	ethereumConn          *ethereum.Connection
 	ethereumChannelWriter *EthereumChannelWriter
 	beefyListener         *BeefyListener
-	log                   *logrus.Entry
 }
 
-const Name = "parachain-relay"
-
-func NewWorker(parachainConfig *parachain.Config,
-	relaychainConfig *relaychain.Config, ethereumConfig *ethereum.Config, log *logrus.Entry) (*Worker, error) {
-
+func NewRelay(config *Config, keypair *secp256k1.Keypair) (*Relay, error) {
 	log.Info("Creating worker")
 
-	ethereumKp, err := secp256k1.NewKeypairFromString(ethereumConfig.MessagePrivateKey)
-	if err != nil {
-		return nil, err
-	}
+	parachainConn := parachain.NewConnection(config.Source.Parachain.Endpoint, nil)
+	relaychainConn := relaychain.NewConnection(config.Source.Polkadot.Endpoint)
 
-	parachainConn := parachain.NewConnection(parachainConfig.Endpoint, nil, log)
-	relaychainConn := relaychain.NewConnection(relaychainConfig.Endpoint, log)
-	ethereumConn := ethereum.NewConnection(ethereumConfig.Endpoint, ethereumKp, log)
+	// FIXME: This is used by both the source & sink. They should use separate connections
```

Comment:
plz use TODO or put in github issues. don't think we've used FIXME anywhere before so it could get lost

Reply:
although if things work reliably with a shared connection, then maybe we don't even have to worry?